Sources/Environment.swift

```diff
@@ -1773,4 +1804,154 @@ class Environment {
     private func doLessThanOrEqual(_ args: [any RuntimeValue]) throws -> Bool {
         return try doLessThan(args) || doEqualTo(args)
     }
+
+    /// Formats a date using strftime-style format specifiers
+    /// - Parameters:
+    ///   - date: The date to format
+    ///   - format: A strftime-compatible format string
+    /// - Returns: The formatted date string
+    static func formatDate(_ date: Date, withFormat format: String) -> String {
```

Comment:
Could you add some test cases for formatDate?

Reply:
Yes I'll add test cases around that.

Reply:
@johnmai-dev Added a new file for date format tests.